src/App.tsx

```diff
@@ -1,9 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, SetQuery] = useState('');
+  const [visibleMovies, SetVisibleMovies] = useState([...moviesFromServer]);
+
+  const filterFilms = (inputValue:string) => {
+    const queryNormilize = inputValue.toLowerCase().trim();
+
+    if (inputValue.length === 0) {
+      SetVisibleMovies([...moviesFromServer]);
+    }
+
+    SetVisibleMovies(moviesFromServer.filter(movie => {
+      const movieTitle = movie.title.toLowerCase();
+      const movieDescription = movie.description.toLowerCase();
+
+      return (movieTitle.includes(queryNormilize)
+          || movieDescription.includes(queryNormilize)
+      );
+    }));
+  };
+
+  useEffect(() => {
+    filterFilms(query);
+  }, [query]);
```

Comment:
I don't understand why you use useEffect in this task, because I fink you don't need that